app/controllers/exercise_solutions_controller.rb

```diff
@@ -22,6 +22,6 @@ def set_messages
   end
 
   def solution_params
-    params.require(:solution).permit(:content)
+    params.require(:solution).tap { |it| it[:content] = params[:solution][:content] }
```

Comment:
What is the point of this change?

Reply:
If you use permit(:content) you need to specify all keys in the hash to permit, but content can be a String or a hash (and hash can have different keys)
Using tap and whitelist, you avoid this problem

Reply:
OK. I got it. Is hacky but I can't think of a better solution, neither. 

Could you please add this explanation in a comment